pageserver/src/task_mgr.rs

```diff
@@ -481,13 +481,20 @@ pub async fn shutdown_tasks(
     for task in victim_tasks {
         let join_handle = {
             let mut task_mut = task.mutable.lock().unwrap();
-            info!("waiting for {} to shut down", task.name);
-            let join_handle = task_mut.join_handle.take();
-            drop(task_mut);
-            join_handle
+            task_mut.join_handle.take()
         };
         if let Some(join_handle) = join_handle {
-            let _ = join_handle.await;
+            tokio::pin!(join_handle);
```

Comment:
NIT: can we just use `if let Some(mut join_handle) = join_handle {` above instead of pinning?

Reply:
Didn't check but most likely.